### PR TITLE
simplified python normalization logic in pca

### PR DESCRIPTION
**Pull request by jbloom22**
- changed normalized_genotype_pca to pca_normalized_genotypes
- improved doc of nGenotypes

### CODE REVIEW

File: python/hail/dataset.py [hunk 1]
```diff
@@ -3408,7 +3408,7 @@ def min_rep(self, left_aligned=False):
     @typecheck_method(k=integral,
                       compute_loadings=bool,
                       as_array=bool)
-    def normalized_genotype_pca(self, k=10, compute_loadings=False, as_array=False):
+    def pca_normalized_genotypes(self, k=10, compute_loadings=False, as_array=False):
```
Comment (cseed) [inline python/hail/dataset.py]: This seems unnecessary.  I also prefer the former.  We don't say "dog big", we say "big dog".

File: python/hail/dataset.py [hunk 2]
```diff
@@ -3425,14 +3425,15 @@ def normalized_genotype_pca(self, k=10, compute_loadings=False, as_array=False):
...
         """
 
-        vds = self.annotate_variants_expr('va.mean = gs.map(g => g.GT.gt).sum() / gs.filter(g => isDefined(g.GT)).count()')\
-                .filter_variants_expr('isDefined(va.mean) && gs.filter(g => isDefined(g.GT)).map(g => g.GT.gt).collectAsSet().size() != 1').persist()
+        vds = (self.annotate_variants_expr('va.AC = gs.map(g => g.GT.gt).sum(), va.nCalled = gs.filter(g => isDefined(g.GT)).count()')
+            .filter_variants_expr('va.AC > 0 && va.AC < 2 * va.nCalled').persist())
         nVariants = vds.count_variants()
         if nVariants == 0:
-            fatal("Cannot run PCA: found 0 variants after filtering out variants with constant genotypes.")
+            fatal("Cannot run PCA: found 0 variants after filtering out variants that are all HomRef or all HomVar.")
```
Comment (cseed) [inline python/hail/dataset.py]: "all homozygous reference or homozygous variant".

File: python/hail/dataset.py [hunk 3]
```diff
@@ -3425,14 +3425,15 @@ def normalized_genotype_pca(self, k=10, compute_loadings=False, as_array=False):
...
         print('Running PCA using ' + str(nVariants) + ' variants.')
-        stddev = 'sqrt(va.mean * (2 - va.mean) * ' + str(nVariants) + ' / 2)'
-        result = vds.pca('if (isDefined(g.GT)) (g.GT.gt - va.mean) / ' + stddev + ' else 0', k, compute_loadings, as_array)
+        result = vds.pca('let mean = va.AC / va.nCalled in ' +
+            'if (isDefined(g.GT)) (g.GT.gt - mean) / sqrt(mean * (2 - mean) * ' + str(nVariants) + ' / 2) else 0',
```
Comment (cseed) [inline python/hail/dataset.py]: I prefer triple quote strings for multi-line string literals.

File: python/hail/dataset.py (hunk 1 above)
Comment (jbloom22) [reply python/hail/dataset.py]: I'm thinking of PCA as a verb like "dimensionally reduce", as it's sometimes used. How about `pca_on_normalized_genotypes` or `pca_with_normalized_genotypes`. All things being equal, it's also nicer for this command to be consecutive with `pca` in docs.

If you feel strongly, I'll change back to `normalized_genotype_pca` but I find that hard to parse.

File: python/hail/dataset.py [hunk 4]
```diff
@@ -3425,14 +3425,17 @@ def normalized_genotype_pca(self, k=10, compute_loadings=False, as_array=False):
...
         if nVariants == 0:
-            fatal("Cannot run PCA: found 0 variants after filtering out variants with constant genotypes.")
-        print('Running PCA using ' + str(nVariants) + ' variants.')
-        stddev = 'sqrt(va.mean * (2 - va.mean) * ' + str(nVariants) + ' / 2)'
-        result = vds.pca('if (isDefined(g.GT)) (g.GT.gt - va.mean) / ' + stddev + ' else 0', k, compute_loadings, as_array)
+            fatal("Cannot run PCA: found 0 variants after filtering out variants that are all homozygous reference or all homozygous variant.")
+        print('Running PCA using {} variants.'.format(nVariants))
+        result = vds.pca("""let mean = va.AC / va.nCalled in
+            if (isDefined(g.GT)) (g.GT.gt - mean) / sqrt(mean * (2 - mean) * {} / 2)
+            else 0""".format(nVariants),
```
Comment (cseed) [inline python/hail/dataset.py]: Double quotes " should be used for human readable messages.  Single quotes ' should be used for machine readable strings/code.  E.g. I'd change the print to double quotes " and this to triple single quotes.

**Comment by jbloom22** [comment]
PCA docs need more revision (e.g., moving GRM specific math from pca to pca_of_normalized_genotypes) but let's leave that to another PR, maybe when we're ready for a docs improvement push as 0.2 stabilizes
